soda_machine.rb

```diff
@@ -7,12 +7,17 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.length
   end
 
   def find_soda(soda_brand)
+    @sodas.find { |soda| soda.brand == soda_brand }
   end
 
   def sell(soda_brand)
+    sold_soda = find_soda(soda_brand)
+    @cash += sold_soda.price if sold_soda
+    @sodas.delete(sold_soda)
```

Comment:
I like the implementation here, cleanly returns the appropriate values based on if the soda is found or not. The only question I can think to raise is do you really want to check the array for the sold_soda if it doesn't exist/is nil?